### PR TITLE
DDF-1174 Adding jacoco config to DDF-Catalog

### PR DESCRIPTION
**Pull request by andrew-fiedler**
The top pom defers to the DDF-Parent configuration, but the rest of the poms have custom configurations for the minimum code coverage required (determined by an initial run with jacoco).

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf-catalog/74)

### CODE REVIEW

**Comment by mrmateo** [comment]
@andrew-fiedler - can you explain the minimum code coverage numbers? It sounds like you ran this manually once and jacoco gave you some output of what the minimums should be for each project?

**Comment by andrew-fiedler** [comment]
Yeah, that's correct.  I wanted to establish a baseline for each project, so it was run manually once.  I subtracted 5 percent from the determinations to give us some room, with a minimum of 0.

**Comment by mrmateo** [comment]
Are you still going to modify this PR after the main parent PR is merged?

**Comment by andrew-fiedler** [comment]
There shouldn't need to be any changes after the main parent PR is merged.  The top level pom already defers configuration to the DDF-Parent pom.

File: core/catalog-core-eventcommands/pom.xml [hunk 1]
```diff
@@ -12,71 +12,115 @@
...
-	
-</project>
\ No newline at end of file
+                            *
+                        </Import-Package>
+                    </instructions>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.jacoco</groupId>
```
Comment (mrmateo) [inline core/catalog-core-eventcommands/pom.xml]: Example from what I was talking about on the parent PR. Is this configuration already in the parent?

**Comment by lessarderic** [comment]
<sup>**[core/catalog-core-eventcommands/pom.xml, line 82 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-Jnw-RpFSa2BRVxy5UE5-r1-82)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/core/catalog-core-eventcommands/pom.xml#L82)):</sup>
Yep. This should be removed.

---

<sup>**[core/catalog-core-impl/filter-proxy/pom.xml, line 73 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-JnxBHsZLxPPJyX4r5DM)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/core/catalog-core-impl/filter-proxy/pom.xml#L73)):</sup>
Uses defaults. Remove.

---

<sup>**[core/catalog-core-impl/metacard-type-registry/pom.xml, line 69 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-JnxBLmJL8A8BPd_ts5D)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/core/catalog-core-impl/metacard-type-registry/pom.xml#L69)):</sup>
Uses defaults. Remove.

---

<sup>**[core/catalog-core-metacardgroomerplugin/pom.xml, line 67 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-JnxBOhNpqdiACnZHo9Q)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/core/catalog-core-metacardgroomerplugin/pom.xml#L67)):</sup>
Uses defaults. Remove.

---

<sup>**[core/catalog-core-resourcesizeplugin/pom.xml, line 100 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-JnxBQqf1tiEn0XDCFhZ)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/core/catalog-core-resourcesizeplugin/pom.xml#L100)):</sup>
Uses defaults. Remove.

---

<sup>**[core/catalog-core-solr/pom.xml, line 271 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-JnxBSryHUEeV6tR2UK8)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/core/catalog-core-solr/pom.xml#L271)):</sup>
Uses defaults. Remove.

---

<sup>**[docs/pom.xml, line 41 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-JnxBWIFPkKdUNex2dsd)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/docs/pom.xml#L41)):</sup>
Uses defaults. Remove.

---

<sup>**[security/catalog-security-filter/pom.xml, line 70 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-JnxBbQuBIkj2PmxMN4r)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/security/catalog-security-filter/pom.xml#L70)):</sup>
Uses defaults. Remove.

---

<sup>**[solr/catalog-solr-embedded-provider/pom.xml, line 276 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-JnxBeha1Ry3jgBwZqWf)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/solr/catalog-solr-embedded-provider/pom.xml#L276)):</sup>
Uses defaults. Remove.

---

<sup>**[solr/docs/pom.xml, line 41 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-JnxBhCGCY2Za1ooeLJs)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/solr/docs/pom.xml#L41)):</sup>
Uses defaults. Remove.

---

<sup>**[transformer/catalog-transformer-geojson-queryresponse/pom.xml, line 71 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-JnxBknhWkQvch7sImkK)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/transformer/catalog-transformer-geojson-queryresponse/pom.xml#L71)):</sup>
Uses defaults. Remove.

---

<sup>**[transformer/catalog-transformer-metadata/pom.xml, line 61 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-JnxBmBRLoREtURzSVVr)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/transformer/catalog-transformer-metadata/pom.xml#L61)):</sup>
Uses defaults. Remove.

---

<sup>**[transformer/catalog-transformer-resource/pom.xml, line 59 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-JnxBneo8UrYuiwzgzG9)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/transformer/catalog-transformer-resource/pom.xml#L59)):</sup>
Uses defaults. Remove.

---

<sup>**[transformer/catalog-transformer-thumbnail/pom.xml, line 59 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-JnxBqTLro_Me-H5y9WZ)** ([raw file](https://github.com/codice/ddf-catalog/blob/96f92997376664d70e88b6f8af1a341ecb83c7b0/transformer/catalog-transformer-thumbnail/pom.xml#L59)):</sup>
Uses defaults. Remove.

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/74)

**Comment by andrew-fiedler** [comment]
@mrmateo Ah, I see.  I'll remove the configurations from the files that are using the defaults (I have a script that can find them).

**Comment by shaundmorris** [comment]
Reviewed files:
- admin/catalog-admin-downloadmanager/pom.xml @ r1
- admin/module/catalog-admin-module-sources/pom.xml @ r1
- admin/plugin/catalog-admin-plugin-sourceconfiguration/pom.xml @ r1
- catalog-app/pom.xml @ r1
- common/catalog-common-geo-formatter/pom.xml @ r1
- core/catalog-core-api-impl/pom.xml @ r1
- core/catalog-core-api/pom.xml @ r1
- core/catalog-core-backupplugin/pom.xml @ r1
- core/catalog-core-camelcomponent/pom.xml @ r1
- core/catalog-core-commands/pom.xml @ r1
- core/catalog-core-commons/pom.xml @ r1
- core/catalog-core-eventcommands/pom.xml @ r2
- core/catalog-core-federationstrategy/pom.xml @ r1
- core/catalog-core-impl/filter-proxy/pom.xml @ r2
- core/catalog-core-impl/metacard-type-registry/pom.xml @ r2
- core/catalog-core-impl/pubsub-tracker/pom.xml @ r1
- core/catalog-core-impl/pubsub/pom.xml @ r1
- core/catalog-core-metacardgroomerplugin/pom.xml @ r2
- core/catalog-core-metricsplugin/pom.xml @ r1
- core/catalog-core-resourcesizeplugin/pom.xml @ r2
- core/catalog-core-solr/pom.xml @ r2
- core/catalog-core-sourcemetricsplugin/pom.xml @ r1
- core/catalog-core-standardframework/pom.xml @ r1
- core/catalog-core-urlresourcereader/pom.xml @ r1
- docs/pom.xml @ r2
- measure/catalog-measure-api/pom.xml @ r1
- opensearch/catalog-opensearch-endpoint/pom.xml @ r1
- opensearch/catalog-opensearch-source/pom.xml @ r1
- plugin/catalog-plugin-federation-replication/pom.xml @ r1
- plugin/catalog-plugin-jpeg2000-thumbnail-converter/pom.xml @ r1
- pom.xml @ r1
- rest/catalog-rest-app/pom.xml @ r1
- rest/catalog-rest-endpoint/pom.xml @ r1
- schematron/catalog-schematron-plugin/pom.xml @ r1
- security/catalog-security-filter/pom.xml @ r2
- security/catalog-security-logging/pom.xml @ r1
- security/catalog-security-plugin/pom.xml @ r1
- solr/catalog-solr-embedded-provider/pom.xml @ r2
- solr/catalog-solr-external-provider/pom.xml @ r1
- solr/docs/pom.xml @ r2
- solr/solr-app/pom.xml @ r1
- transformer/catalog-transformer-geojson-input/pom.xml @ r1
- transformer/catalog-transformer-geojson-metacard/pom.xml @ r1
- transformer/catalog-transformer-geojson-queryresponse/pom.xml @ r2
- transformer/catalog-transformer-metadata/pom.xml @ r2
- transformer/catalog-transformer-resource/pom.xml @ r2
- transformer/catalog-transformer-service-atom/pom.xml @ r1
- transformer/catalog-transformer-service-xslt/pom.xml @ r1
- transformer/catalog-transformer-thumbnail/pom.xml @ r2
- transformer/catalog-transformer-tika-input/pom.xml @ r1
- transformer/catalog-transformer-xml-binding/pom.xml @ r1
- transformer/catalog-transformer-xml/pom.xml @ r1

---

<sup>**[pom.xml, line 391 \[r2\]](https://reviewable.io:443/reviews/codice/ddf-catalog/74#-Jo5oyAvuwgdViwZm5V2)** ([raw file](https://github.com/codice/ddf-catalog/blob/909d7d3764758a77f96b7ece6583901884bf766a/pom.xml#L391)):</sup>
@andrewkfiedler do you ned the surefire plugin here or could it be inherited from ddf-parent?

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/74)

**Comment by shaundmorris** [comment]
The Travis build was failing due to https://tools.codice.org/jira/browse/DDF-1176. I'll go ahead and address the merge conflict and my last comment :-)

---









Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/74)

**Pull request merged**